snetd/cmd/flags.go

```diff
@@ -0,0 +1,48 @@
+package cmd
+
+import (
+	"github.com/singnet/snet-daemon/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cfgFile            = ServeCmd.PersistentFlags().StringP("config", "c", "snetd.config", "config file")
+	daemonType         = ServeCmd.PersistentFlags().StringP("type", "t", "grpc", "daemon type: one of 'grpc','http'")
+	blockchainEnabled  = ServeCmd.PersistentFlags().BoolP("blockchain", "b", true, "enable blockchain processing")
+	listenPort         = ServeCmd.PersistentFlags().IntP("port", "p", 5000, "daemon listen port")
+	ethEndpoint        = ServeCmd.PersistentFlags().String("ethereum-endpoint", "http://127.0.0.1:8545", "ethereum JSON-RPC endpoint")
+	mnemonic           = ServeCmd.PersistentFlags().String("mnemonic", "", "HD wallet mnemonic")
+	hdwIndex           = ServeCmd.PersistentFlags().Int("wallet-index", 0, "HD wallet index")
+	dbPath             = ServeCmd.PersistentFlags().String("db-path", "snetd.db", "database file path")
+	passthroughEnabled = ServeCmd.PersistentFlags().Bool("passthrough", false, "passthrough mode")
+	serviceType        = ServeCmd.PersistentFlags().String("service-type", "grpc", "service type: one of 'grpc','jsonrpc','process'")
+	wireEncoding       = ServeCmd.PersistentFlags().String("wire-encoding", "proto", "message encoding: one of 'proto','json'")
+	pollSleep          = ServeCmd.PersistentFlags().String("poll-sleep", "5s", "blockchain poll sleep time")
+)
+
+func init() {
+	rf := ServeCmd.PersistentFlags()
+	vip := config.Vip()
+
+	vip.BindPFlag(config.ConfigPathKey, rf.Lookup("config"))
```

Comment:
Maybe we should rethink this config key name & associated variable names (currently referred to as "path" and "file" in various places). In reality, this populates with vip expects to be the filename prefix (without `.json`). There is also a path to search in, which is currently hardcoded as cwd.

Reply:
oh, good point, this isn't really a "path" at all, it's a filename. we should definitely change these flags.

would you like to see that in this PR, or a separate one?

Reply:
I'd say this one for less PR noise & nothing gained from pushing intermediate code separately

Reply:
Done